imap_processing/hi/hi_cdf_attrs.py

```diff
@@ -50,87 +33,102 @@
     validmin=0,
     validmax=10,
     format="I2",
-    label_axis="ESA step",
+    label_axis="Energy Step",
     display_type="time_series",
     catdesc=(
-        "ESA step. "
+        "ESA step (0-10), nominally 9, but possibly 8 or 10,"
+        " 0 is likely a background test"
+    ),
+    var_notes=(
+        "ESA (electrostatic analyzer) step. "
         "It's a 4-bits integer value that represents the "
         "ESA step. ESA step value range from 0-10. "
         "nominally 9 ESA steps, but possibly 8 or 10. "
-        "'Step 0' is likely to refer to a background test."
+        "'Step 0' is likely to refer to a background test. "
+        "This value is used to look up the actual energy "
+        "value from its lookup table."
     ),
-    fieldname="ESA step",
+    fieldname="ESA step number (0-10)",
     fill_val=GlobalConstants.INT_FILLVAL,
-    var_type="metadata",
+    var_type="support_data",
     depend_0="epoch",
 )
 
 de_tag_attrs = ScienceAttrs(
-    validmin=GlobalConstants.INT_FILLVAL,
-    validmax=GlobalConstants.INT_MAXVAL,
-    format="I6",
-    label_axis="Direct event time tag",
+    validmin=0,
+    validmax=GlobalConstants.UBIT16_MAXVAL,
```

Comment:
I _think_ this validmax means the global max and shouldn't be changing across files, but I really don't know.

My suggestion was basically to infer the datatype rather than manually input it here if that is possible. It likely isn't worth doing here, was more just a thought on possibilities.

Reply:
Oh ok. That's what I was doing with `UINT32_MAXVAL` to set global max since I know that de_tag won't have value great than 16-bits int value. We can discuss that as team and figure out how we want to set global max.

Reply:
I tried to incorporate `dtype` in the science data. Can you let me know what you think or is it what you are thinking?

Reply:
Yes, that is nice :) and it gets me thinking that we should actually check other instruments and datafiles to make sure we are using the correct dtypes, because the default is `int64` and if you only need `uint8` then we are wasting quite a bit of space on disk.

```python
xr.DataArray([0, 1, 2]).dtype  # int64
xr.DataArray(np.asarray([0, 1, 2], np.uint8)).dtype  # uint8
```

Reply:
Totally agree. We should do that in upcoming PR. I can help with that.